client/http-client/src/transport.rs

```diff
@@ -87,7 +92,18 @@ impl HttpTransportClient {
 						.build(),
 					_ => return Err(Error::InvalidCertficateStore),
 				};
-				HyperClient::Https(Client::builder().build::<_, hyper::Body>(connector))
+				match proxy.clone() {
+					Some(pr) => {
+						let proxy_uri = pr.parse().unwrap();
+						let proxy_obj = Proxy::new(Intercept::All, proxy_uri);
+						let proxy_connector = ProxyConnector::from_proxy(connector, proxy_obj).unwrap();
```

Comment:
we probably need to add `IoError` for this to work but no unwraps are allowed here.

```suggestion
						let proxy_connector = ProxyConnector::from_proxy(connector, proxy_obj)?;
```